modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java

```diff
@@ -462,6 +464,55 @@ private DynamicQuery _getSegmentsExperienceIdsDynamicQuery(
 		return dynamicQuery;
 	}
 
+	private void _publishSegmentsExperienceVariant(
+		SegmentsExperience controlSegmentsExperience,
+		SegmentsExperience variantSegmentsExperience) {
+
+		int lowestSegmentsExperiencePriority = Optional.ofNullable(
+			segmentsExperiencePersistence.fetchByG_C_C_Last(
+				controlSegmentsExperience.getGroupId(),
+				controlSegmentsExperience.getClassNameId(),
+				controlSegmentsExperience.getClassPK(), null)
+		).map(
+			SegmentsExperience::getPriority
+		).orElse(
+			SegmentsExperienceConstants.PRIORITY_DEFAULT
+		);
+
+		int controlSegmentsExperiencePriority =
+			controlSegmentsExperience.getPriority();
+		int variantSegmentsExperiencePriority =
+			variantSegmentsExperience.getPriority();
+
+		controlSegmentsExperience.setPriority(
+			lowestSegmentsExperiencePriority - 1);
```

Comment:
You should be very carefull with lowestSegmentsExperiencePriority - 1, because we now have the 0 and -2 problem

Reply:
I know, I'm using the same code as in the SegmentsExperienceLocalService. Do you have any extra problem issue comparing with the previous service?

Reply:
Sure, because you are calling the update method directly, no the update priority method. This method update directly, witouth taking into account any consideration.

So if you create a test per case, you will find many bugs.

Reply:
I'm going to take a look, but this set is only to swap the values temporarily, I mean, It's impossible (or at least I can't see) that the  experiences involved would have this issue, because we're just swapping their priorities